modular_skyrat/modules/xenoarch/code/modules/research/xenoarch/xenoarch_machine.dm

```diff
@@ -7,19 +7,31 @@
 	use_power = IDLE_POWER_USE
 	idle_power_usage = 100
 	pass_flags = PASSTABLE
-	/// the item that holds everything
-	var/obj/item/storage_unit
+	/// Xenoarch-related things we're currently storing
+	var/list/obj/item/xenoarch_contents = list()
```

Comment:
why not just a list? is this even valid?

Reply:
it is a list, it just holds references to every xenoarch item inserted into the machine so it can be iterated through without catching stuff like the machine circuit board

Reply:
the fucked up /obj/item is the old implementation